### ISSUE TITLE
Global search input is not shown

### ISSUE THREAD

**Issue by WillyReyno** [laravel/nova-issues#4022]
- Laravel Version: 8.83.8
- Nova Version: 4.2.2
- PHP Version: 8.0
- Database Driver & Version: MySQL 8.0

### Description:

The global search input is not visible anymore since upgrading from 3 to 4.

![Capture d’écran 2022-04-17 à 19 01 50](https://user-images.githubusercontent.com/5576482/163727309-1d08994e-a4ee-419f-87f7-0a33f0774d3c.png)

I first created a discussion #4021 and decided to open an issue since I wasn't the only one noticing this.

### Detailed steps to reproduce the issue on a fresh Nova installation:
- Open Nova
- Notice the global search input does not appear in the navbar

Feel free to tell me if there's any relevant info I can give to help you debug it.

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository.

**Comment by crynobone** [comment]
Are you guys somehow calling `Nova::provideToScript()` outside of `Nova::serving()`?

**Comment by WillyReyno** [comment]
I use `Nova::provideToScript()` twice **inside** `Nova::service()` : 

1) Inside a custom tool : 
```
Nova::serving(function (ServingNova $event) {
            $service = new Service();
            $service->init();
            Nova::provideToScript([
                'developerToken' => $service->api()->getDeveloperToken(),
            ]);
        });
```

2) Inside a custom Card: 
```
Nova::serving(function (ServingNova $event) {
            Nova::provideToScript([
                'events' => Cache::get('planning'),
            ]);

            Nova::script('planning-card', __DIR__.'/../dist/js/card.js');
            Nova::style('planning-card', __DIR__.'/../dist/css/card.css');
        });
```

But I confirm that commenting both `provideToScript` makes the global search appear again...

**Comment by wdelfuego** [comment]
I'm also setting up a new custom tool locally and only call `provideToScript` once, within the `Nova::serving` call in the `boot` method of my ToolServiceProvider. Commenting out the `provideToScript` call makes global search bar appear. 

Calling `provideToScript` with an empty array makes the global search bar disappear as well.

**Comment by crynobone** [comment]
This doesn't happened on Nova 4 fresh install. I feel we still missing some important information here

**Comment by wdelfuego** [comment]
Did some debugging in `src/Nova.php`; 
```
    public static function provideToScript(array $variables)
    {
        if (empty(static::$jsonVariables)) {
            $userId = Auth::guard(config('nova.guard'))->id() ?? null;

            static::$jsonVariables = [
                'logo' => static::logo(),
                'brandColors' => config('nova.brand.colors', []),
                'hasGloballySearchableResources' => static::hasGloballySearchableResources(),
```
On the last line, `static::hasGloballySearchableResources` returns false when I enable the `provideToScript` call in my ToolServiceProvider. If I hardcode `true` here, the global search bar shows even with the `provideToScript` call enabled.

**Comment by wdelfuego** [comment]
Has this issue been fixed or was it closed because of lack of reproducibility?

**Comment by wdelfuego** [comment]
Looking at the releases page; looks like it’s fixed, awesome!

@davidhemphill is it really too much to ask to respond to simple questions of us paying users? You could have been the bringer of good news just saying “fixed” and we’d all be happy. This silence treatment we get really baffles me.

**Comment by crynobone** [comment]
@wdelfuego It's 5.25am where I'm at right now, please be reasonable with response time expectation.

Issue with `bug` label and assigned will (and can only) be automatically closed once the relevant bugfix PR has been merged. We don't wait for the patch release to close tickets.

**Comment by wdelfuego** [comment]
This has nothing to do with response time expectation, this has to do with communication plain and simple. 

Silently closing an issue in which the last update was “can’t reproduce, need more information” is worrying. Other bug reports have been closed without further notice or explanation before and specific requests for information directed at @davidhemphill have been ignored consistently and indefinitely. My email to the support team has gone unanswered too.

Just adding a comment “fixed” would tell us why this issue has been closed and that we don’t have to investigate further. 

This is not about response time, this is about quality of communication.

**Comment by davidhemphill** [comment]
@wdelfuego It was closed because a Pull Request that uses "Fixed laravel/nova-issues#4022" (which automatically closes the issue) was merged. I saw your comment (https://github.com/laravel/nova-issues/issues/4022#issuecomment-1104154179), but since the pull request was merged and we intend to do a release today...I did not.

Issues that are not real bug requests will typically receive a response. In this case, @crynobone found the issue on his own and submitted a pull request...we merged it and the issue was closed automatically. 

Like @crynobone said, please be reasonable with your response time expectations. Also, a little honey goes a long way. You're rather confrontational on here and I would rather refund your license than continue seeing your pushy comments all over here. 🫤

**Comment by wdelfuego** [comment]
Thanks, David!

> we merged it and the issue was closed automatically.

That's good to know! GitHub says "davidhemphill closed this 15 hours ago" which makes it look like a manual action, hence my question.

> I saw your comment (https://github.com/laravel/nova-issues/issues/4022#issuecomment-1104154179), but since the pull request was merged and we intend to do a release today...I did not.

That's the type of communication decision I don't understand. Was that request for information problematic to you? It was meant as a neutral request for information that could be answered in two words. I apologize if it seemed something else to you. @crynobone had no issues [responding](https://github.com/laravel/nova-issues/issues/3928#issuecomment-1096162573) to a question like that before.

I like pushyness as little as you do but I don't know what to do when being friendly means to be ignored.

Look at the issue about [DateTime formatting](https://github.com/laravel/nova-issues/discussions/3929) where all of the following comments (by me and other users) were completely ignored:

> Could you let us know your stance on the issue of formatting DateTime fields on Index and Detail views? 

> I would like to know if the Nova team agrees that this should be considered a bug and if there is any indication as to when the feature will be back in Nova 4. Thanks!

> I hope we agree that there is still at the very least a documentation issue

> in my opinion, it shouldn't be this hard to discover how to make a sortable column of formatted DateTime fields in a paid-for admin panel

>  it's completely understandable that Nova 4 is very new and that we have to discover these types of things together. Thanks for putting me in the right direction

> So we lost the plain and simple DateTime format method to this? [...] This makes me all the more curious to know the reasons behind this decision.

If those aren't friendly enough requests for information I don't really know how to go forward interacting with the maintainers in a way that works for you. A simple "we consider this a bug and hope to fix it soon" or "we don't consider this a bug so build a custom solution" would have been super helpful -- let's keep the honey flowing in both directions.

Nova is a wonderful product that is very much worth its money. I'm a very enthusiastic user trying to contribute to making it better and I'm investing loads of time into making it perfect for my clients. Any pushyness you perceive comes from wanting the product and the communication surrounding its development to be (even) better and effective. If you'd rather I stop being active on this issues forum altogether you only have to say so and I will figure things out on my own from now on.

Thanks for a wonderful product and thanks a lot for responding to this criticism, it really means a lot!
